e2e/controls.py

```diff
@@ -114,3 +166,128 @@ def locate_by_label(self, label: str) -> Locator:
 
     def locate_by_value(self, value: str) -> Locator:
         return self.loc.locator("input[type=radio][value={value}]")
+
+class SelectInput(SimpleInput):
+    def __init__(self, page: Page, inputId: str):
+        super().__init__(page, f"select#{inputId}.shiny-bound-input")
+
+    # TODO
+    # def get_options(self):
+
+    def get_selected(self):
+        return self.loc.locator("option[selected]").inner_text()
+
+    def select_option(self, value: str):
+        self.loc.select_option(value)
+
+class SelectizeInput():
+    def __init__(self, page: Page, inputId: str):
+        self.page = page
+        self.loc = page.locator(f"//*[@id='{inputId}']//ancestor::div[contains(@class, 'shiny-input-container')]")
+
+    def get_selected_items(self) -> List:
+        selected_items = self.loc.locator(".selectize-input .item")
+        mylist = []
+        for i in range(selected_items.count()):
+            mylist.append(selected_items.nth(i).inner_text())
+
+        return mylist
+
+    def select_option(self, value: str):
+        self.loc.locator(".selectize-input").click()
+
+        dropdown = self.loc.locator(".selectize-dropdown")
+        expect(dropdown).to_be_visible()
+
+        self.loc.locator(f".selectize-dropdown-content .option[data-value='{value}']").click()
+
+        self.loc.locator(f".selectize-input .item[data-value='{value}']")
+
+        # click outside the dropdown box to close the dropdown menu
+        self.page.keyboard.press("Escape")
+
+    #TODO
+    #def remove_option(self):
+
+
+class FileInput():
+    def __init__(self, page: Page, inputId: str):
+        self.loc = page.locator(f"//*[@id='{inputId}']//parent::div[contains(@class, 'shiny-input-container')]")
+
+    def upload_file(self, fileName: str) -> None:
+        browse = self.loc.locator("//span[contains(@class, 'btn-file')]")
+        #TODO: Enhancement: Check PurePath option to upload files
+        browse.set_input_files(files=[f"e2e/data-files/{fileName}"])
+
+# Shiny Outputs
+class TextOutput():
+    def __init__(self, page: Page, inputId: str):
+        self.loc = page.locator(f"#{inputId}.shiny-text-output.shiny-bound-output")
+
+    def get_text(self) -> str:
+        return self.loc.inner_text()
+
+    def contains_digit(self, text: str) -> bool:
```

Comment:
This doesn't belong on the TextOutput class, since it doesn't relate to the TextOutput instance. Also, this searches to see if _any_ character in the text is a digit, is this really what you want? (versus testing to see if the entire string is some kind of number?)

Reply:
Thanks for bringing this up. Yes, my thought was to check that the textoutput shows the "Latitude" and "Longitude" of the ipyleaflet map. Test app using this function is https://github.com/rstudio/py-shiny/pull/355/files#diff-592926b6d35d2782ae85d50863febd3c90255d3da656ea16db327ecc8b9481e1R38.

So, I am checking if that string contains atleast one digit. we are good to believe that map coordinates are shown in the output.as 

However, I know this is not a good approach and has a TODO to fix. I will think about what you suggested and implement this in a way that we dont have to use `assert` also as you mentioned in the later comments.